core/camera/src/main/java/com/google/jetpackcamera/core/camera/CameraXCameraUseCase.kt

```diff
@@ -197,6 +215,87 @@ constructor(
+            cameraManager.getCameraCharacteristics(logicalCameraId).physicalCameraIds) {
+                logicalCameraData.put(
+                    "physical-$physicalCameraId",
+                    getCameraPropertiesJSONObject(physicalCameraId, cameraManager)
+                )
+            }
+            result.put(logicalCameraData)
+        }
+        return result
+    }
+
+    private fun getCameraPropertiesJSONObject(
+        cameraId: String,
+        cameraManager: CameraManager
+    ): JSONObject {
+        val cameraCharacteristics =
+            cameraManager.getCameraCharacteristics(cameraId)
+        val jsonObject = JSONObject()
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            cameraCharacteristics.get(LENS_POSE_ROTATION)
+                ?.let { jsonObject.put(LENS_POSE_ROTATION.name, it) }
+            cameraCharacteristics.get(LENS_POSE_TRANSLATION)
+                ?.let { jsonObject.put(LENS_POSE_TRANSLATION.name, it) }
+            cameraCharacteristics.get(LENS_INTRINSIC_CALIBRATION)
+                ?.let { jsonObject.put(LENS_INTRINSIC_CALIBRATION.name, it) }
+        }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            cameraCharacteristics.get(LENS_DISTORTION)
+                ?.let { jsonObject.put(LENS_DISTORTION.name, it) }
+        }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            cameraCharacteristics.get(CONTROL_ZOOM_RATIO_RANGE)
+                ?.let { jsonObject.put(CONTROL_ZOOM_RATIO_RANGE.name, it) }
+        }
+        cameraCharacteristics.get(LENS_INFO_AVAILABLE_FOCAL_LENGTHS)
+            ?.let { jsonObject.put(LENS_INFO_AVAILABLE_FOCAL_LENGTHS.name, it) }
+        cameraCharacteristics.get(LENS_INFO_MINIMUM_FOCUS_DISTANCE)
+            ?.let { jsonObject.put(LENS_INFO_MINIMUM_FOCUS_DISTANCE.name, it) }
+        cameraCharacteristics.get(REQUEST_AVAILABLE_CAPABILITIES)
+            ?.let { jsonObject.put(REQUEST_AVAILABLE_CAPABILITIES.name, it) }
+
+        return jsonObject
+    }
+
+    private fun writeFileExternalStorage(file: File, textToWrite: String) {
+        //Checking the availability state of the External Storage.
+        val state = Environment.getExternalStorageState()
+        if (Environment.MEDIA_MOUNTED != state) {
+            //If it isn't mounted - we can't write into it.
+
+            return
+        }
+
+        //This point and below is responsible for the write operation
+        var outputStream: FileOutputStream? = null
+        try {
+            file.createNewFile()
+            //second argument of FileOutputStream constructor indicates whether
+            //to append or create new file if one exists
+            outputStream = FileOutputStream(file, true)
+
+            outputStream.write(textToWrite.toByteArray())
+            outputStream.flush()
+            outputStream.close()
+        } catch (e: java.lang.Exception) {
+            e.printStackTrace()
```

Comment:
Don't use `e.printStackTrace()`. Instead use `Log.e(TAG, <message>, <exception>)`.

However, we should also consider whether we want to catch the exception here or just have it thrown and crash the app. If we're in debug mode, we're expecting the file to be written, so I would lean towards just throwing the exception and allowing it to crash the app.